File: Python/Etc/가장 짧은 문자거리.py
word, n = map(str, input().split(" "))

data = [0 for _ in range(len(word))]

temp = []

for i in range(len(word)):
    if word[i] == n:
        temp.append(i)

check = 0
for i in range(len(temp)):
    te = temp[i]

    for j in range(len(word)):
        if j < te:
            if check == 0:
                data[j] = te - j
            else:
                if data[j] > te - j:
                    data[j] = te - j
        elif j > te:
            if check == 0:
                data[j] = j - te
            else:
                if data[j] > j - te:
                    data[j] = j - te
        else:
            data[j] = 0
    check = 1

for n in data:
    print(n, end=' ')

File: Java/Etc/가장 짧은 문자거리.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    public int[] solution(String str, char n) {
        int[] data = new int[str.length()]; // 다 0으로 초기화됨 ! 그게 기본!
        ArrayList<Integer> temp = new ArrayList<>();

        for (int i = 0 ; i < str.length() ; i++) {
            if (str.charAt(i) == n) {
                temp.add(i);
            }
        }
        int check = 0;
        for (int num : temp) {
            for (int j = 0; j < str.length(); j++) {
                if (num > j) {
                    if (check == 0) {
                        data[j] = num - j;
                    } else {
                        if (data[j] > num - j) {
                            data[j] = num - j;
                        }
                    }
                } else if (num < j) {
                    if (check == 0) {
                        data[j] = j - num;
                    } else {
                        if (data[j] > j - num) {
                            data[j] = j - num;
                        }
                    }
                } else {
                    data[j] = 0;
                }
            }
            check = 1;
        }
        return data;
    }

    public int[] solution2 (String str, char n) {
        int[] data = new int[str.length()];
        int p = 10000;
        for (int i = 0 ;i < str.length() ; i++) {
            if (str.charAt(i) == n) p = 0;
            else {

                p++;
                data[i] = p;
            }
        }
        p = 10000;
        for (int i = str.length()-1 ; i >= 0 ; i--) {
            if (str.charAt(i) == n) p = 0;
            else {
                p++;
                data[i] = Math.min(p, data[i]);
            }
        }
        return data;
    }
    public static void main(String[] args) {
        Main T = new Main();
        Scanner scanner = new Scanner(System.in);
        String word = scanner.next();
        char n = scanner.next().charAt(0);
        for(int x : T.solution2(word, n)) {
            System.out.print(x+ " ");
        }
    }

}
